core/test/unit/api/canary/utils/validators/input/pages_spec.js

```diff
@@ -137,49 +137,49 @@ describe('Unit: canary/utils/validators/input/pages', function () {
         });
 
         describe('field formats', function () {
-            const fieldMap = {
-                title: [123, new Date(), _.repeat('a', 2001)],
-                slug: [123, new Date(), _.repeat('a', 192)],
-                mobiledoc: [123, new Date()],
-                feature_image: [123, new Date(), 'random words'],
-                featured: [123, new Date(), 'abc'],
-                status: [123, new Date(), 'abc'],
-                locale: [123, new Date(), _.repeat('a', 7)],
-                visibility: [123, new Date(), 'abc'],
-                meta_title: [123, new Date(), _.repeat('a', 301)],
-                meta_description: [123, new Date(), _.repeat('a', 501)]
-            };
-
-            Object.keys(fieldMap).forEach((key) => {
-                it(`should fail for bad ${key}`, function () {
-                    const badValues = fieldMap[key];
-
-                    const checks = badValues.map((value) => {
-                        const page = {};
+            let fieldMap, badValues, checks, page, key, frame;
+            before(function () {
+                fieldMap = {
+                    title: [123, new Date(), _.repeat('a', 2001)],
+                    slug: [123, new Date(), _.repeat('a', 192)],
+                    mobiledoc: [123, new Date()],
+                    feature_image: [123, new Date(), 'random words'],
+                    featured: [123, new Date(), 'abc'],
+                    status: [123, new Date(), 'abc'],
+                    locale: [123, new Date(), _.repeat('a', 7)],
+                    visibility: [123, new Date(), 'abc'],
+                    meta_title: [123, new Date(), _.repeat('a', 301)],
+                    meta_description: [123, new Date(), _.repeat('a', 501)]
+                };
+            });
+           
+            beforeEach(function () {
+                Object.keys(fieldMap).forEach((key) => {
+                    badValues = fieldMap[key];
+                    checks = badValues.map((value) => {
+                        page = {};
                         page[key] = value;
-
                         if (key !== 'title') {
                             page.title = 'abc';
                         }
-
-                        const frame = {
+                        frame = {
                             options: {},
                             data: {
                                 pages: [page]
                             }
                         };
-
                         return validators.input.pages.add(apiConfig, frame)
                             .then(Promise.reject)
                             .catch((err) => {
                                 (err instanceof common.errors.ValidationError).should.be.true();
                             });
                     });
-
-                    return Promise.all(checks);
                 });
+            });    
+            it(`should fail for bad ${key}`, function () {
```

Comment:
This function would always return `should fail for bad undefined`. The idea was to have a checked key returned for each iteration using `Object.keys(fieldMap).forEach((key) => {` :wink: